kore/src/Kore/Step/Simplification/Exists.hs

```diff
@@ -198,16 +199,16 @@ makeEvaluate
     variable
     patt@Predicated { term, predicate, substitution }
   =
-    case localSubstitution of
-        [] ->
+    case boundSubstitution of
+        Nothing ->
             return (makeEvaluateNoFreeVarInSubstitution variable patt)
-        _ -> do
+        Just boundTerm -> do
             (substitutedPat, _proof) <-
                 substituteTermPredicate
                     term
                     predicate
-                    localSubstitutionList
-                    (Substitution.wrap globalSubstitution)
+                    (Map.singleton variable boundTerm)
+                    (Substitution.unsafeWrap $ Map.toList freeSubstitution)
```

Comment:
This step will only work if the substitution is normalized, otherwise we can't apply the substitution correctly. Should we call `normalize` here to ensure that it is normalized, or fall back to the second case if it is not?

Reply:
I'm worried that there is no simple and obvious way (preferably one that's also stable through code changes) to see that `freeSubstitution` is, indeed, normalized.

Calling `normalize` here should be fine. Using `Kore.Unification.Substitution.partition` or something close to it (but declared in the same module) that produces a wrapped substitution would also be fine.